llvm/lib/Target/AIE/AIEWawRegRewriter.cpp

```diff
@@ -137,6 +149,27 @@ class AIEWawRegRewriter : public MachineFunctionPass {
   /// removed for future replacement strategies, i.e. block wl4, wh4, y2 if X4
   /// is used.
   void addAliasRegs(BitVector &BlockedPhysRegs, const MCPhysReg PhysReg) const;
+
+  /// \return additional blocked registers of the \p MF
+  PhysRegVec getAdditionalBlockedRegs(const MachineFunction &MF) const;
+
+  /// Accumulate defined registers within the \param MBB in \param UsedRegs
+  void accumulateDefRegs(const MachineBasicBlock &MBB,
+                         std::set<MCPhysReg> &UsedRegs) const;
+
+  /// Return all the GPR and alias register within \p MF
+  PhysRegVec getGPRAndAliasRegs(const MachineFunction &MF) const;
+
+  /// Add Register to \p UsedRegs, if all its alias register are already used
+  /// and  exist in \p AdditionalUsedRegs , i.e. if r16 and r17 are already
+  /// used, also insert their combined alias register (l0) into \p UsedRegs
+  /// since it is consequently also used.
+  /// Fixme: Work on Regunits and remove this method
+  void setAliasUsage(std::set<MCPhysReg> &UsedRegs,
```

Comment:
This seems different from adding a plain GPR renaming capability. Can that be in a separate commit?